Unable to build unittests on Windows

**Describe the bug/issue**

When trying to build the unittests [src/tests/unittests](..blob/master/src/tests/unittests) on Windows, I get a linker-error that the reference to 'wWinMain' is undefined:
````
$ make test_sample
...
[100%] Linking C executable test_sample.exe
C:/tools/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:/tools/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/lib/../lib/libmingw32.a(lib64_libmingw32_a-crt0_w.o): in function `wmain':
C:/_/M/mingw-w64-crt-git/src/mingw-w64/mingw-w64-crt/crt/crt0_w.c:23: undefined reference to `wWinMain'
collect2.exe: error: ld returned 1 exit status
make[3]: *** [bin/tests/unittests/CMakeFiles/test_sample.dir/build.make:98: bin/tests/unittests/test_sample.exe] Error 1
make[2]: *** [CMakeFiles/Makefile2:7411: bin/tests/unittests/CMakeFiles/test_sample.dir/all] Error 2
make[1]: *** [CMakeFiles/Makefile2:7418: bin/tests/unittests/CMakeFiles/test_sample.dir/rule] Error 2
make: *** [Makefile:3267: test_sample] Error 2
````

Any ideas what I am missing?

I'm not too familiar with the mingw toolchain, so maybe there is a very simple solution and this is more of a layer-8-problem :)


**To Reproduce**

I'm building darktable on windows following the instructions in  [packaging/windows/BUILD.md](../blob/master/packaging/windows/BUILD.md)
-> darktable builds successfully

In order to activate unittests, I followed / adapted the instructions in [src/tests/unittests/README.md](../blob/master/src/tests/unittests/README.md):
Install package
```
pacman -S mingw-w64-x86_64-cmocka
```

Configure / Build via CMake:
```
mkdir build
cd build 
cmake -G "MSYS Makefiles" -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=/opt/darktable -DBUILD_TESTING=ON ../.
make test_sample
````


**Expected behavior**

I can build the tests


**Platform**

* darktable version : current master-branch
* OS                : Win10
* Build-Toolchain: MSYS2 , [msys.txt](https://github.com/darktable-org/darktable/files/6950002/msys.txt)
* mingw-w64-x86_64-gcc 10.3.0-5

First idea was to include the main-wrapper also used in the other main methods - to no avail:
````
#ifdef _WIN32
#include "win/main_wrapper.h"
#endif

...
int main(int argc, char* argv[])
````

Message:
````
$ cmake --build .
...
C:/tools/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles/test_sample.dir/objects.a(test_sample.c.obj): in function `wmain':
C:/work/darktable/darktable/src/win/main_wrapper.h:12: undefined reference to `g_malloc0'
C:/tools/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:/work/darktable/darktable/src/win/main_wrapper.h:14: undefined reference to `g_utf16_to_utf8'
C:/tools/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:/work/darktable/darktable/src/win/main_wrapper.h:16: undefined reference to `g_strfreev'
collect2.exe: error: ld returned 1 exit status
make[2]: *** [bin/tests/unittests/CMakeFiles/test_sample.dir/build.make:98: bin/tests/unittests/test_sample.exe] Error 1
make[1]: *** [CMakeFiles/Makefile2:7411: bin/tests/unittests/CMakeFiles/test_sample.dir/all] Error 2

````

--- a/packaging/windows/BUILD.md
+++ b/packaging/windows/BUILD.md
@@ -25,6 +25,11 @@ How to make a darktable windows installer (64 bit only):
     $ pacman -S mingw-w64-x86_64-gmic
     ```
 
+* Optional: Install libraries required for [testing](../../src/tests/unittests/README.md)
+    ```
+    $ pacman -S mingw-w64-x86_64-cmocka
+    ```
+
 * For gphoto2:
     * You need to restart the MINGW64 or MSYS terminal to have CAMLIBS and IOLIBS environment variables properly set for LIBGPHOTO are available.
     * make sure they aren't pointing into your normal windows installation in case you already have darktable installed. You can check them with:

--- a/src/tests/unittests/CMakeLists.txt
+++ b/src/tests/unittests/CMakeLists.txt
@@ -3,3 +3,8 @@ add_subdirectory(iop)
 add_cmocka_test(test_sample
                 SOURCES test_sample.c
                 LINK_LIBRARIES cmocka)
+
+# Windows: main-method requires the wrapper provided by lib-darktable
+if(WIN32)
+    target_link_libraries(test_sample PRIVATE lib_darktable)
+endif(WIN32)

--- a/src/tests/unittests/iop/test_filmicrgb.c
+++ b/src/tests/unittests/iop/test_filmicrgb.c
@@ -35,6 +35,10 @@
 
 #include "iop/filmicrgb.c"
 
+#ifdef _WIN32
+#include "win/main_wrapper.h"
+#endif
+
 /*
  * DEFINITIONS
  */
@@ -499,11 +503,10 @@ static void test_linear_saturation(void **state)
   testimg_free(ti);
 }
 
-
 /*
  * MAIN FUNCTION
  */
-int main()
+int main(int argc, char* argv[])
 {
   const struct CMUnitTest tests[] = {
     cmocka_unit_test(test_name),

--- a/src/tests/unittests/test_sample.c
+++ b/src/tests/unittests/test_sample.c
@@ -23,12 +23,16 @@
 
 #include <cmocka.h>
 
+#ifdef _WIN32
+#include "win/main_wrapper.h"
+#endif
+
 static void test_sample(void **state)
 {
   // most efficient test: does just nothing :-)
 }
 
-int main()
+int main(int argc, char* argv[])
 {
   const struct CMUnitTest tests[] = {
     cmocka_unit_test(test_sample)

--- a/src/tests/variables.c
+++ b/src/tests/variables.c
@@ -3,6 +3,10 @@
 
 #include <stdio.h>
 
+#ifdef _WIN32
+#include "win/main_wrapper.h"
+#endif
+
 typedef struct test_case_t
 {
   char *input, *expected_result;
@@ -193,13 +197,13 @@ static const test_t test_real_paths = {
     printf("%d / %d tests failed\n\n", n_failed, n_tests);\
 }
 
-int main()
+int main(int argc, char* argv[])
 {
-  char *argv[] = {"darktable-test-variables", "--library", ":memory:", "--conf", "write_sidecar_files=FALSE", NULL};
-  int argc = sizeof(argv) / sizeof(*argv) - 1;
+  char *argv_override[] = {"darktable-test-variables", "--library", ":memory:", "--conf", "write_sidecar_files=FALSE", NULL};
+  int argc_override = sizeof(argv_override) / sizeof(*argv_override) - 1;
 
   // init dt without gui and without data.db:
-  if(dt_init(argc, argv, FALSE, FALSE, NULL)) exit(1);
+  if(dt_init(argc_override, argv_override, FALSE, FALSE, NULL)) exit(1);
 
   int n_tests_overall = 0, n_failed_overall = 0, n_test_functions = 0, n_test_functions_failed = 0;
 

